picard/tagger.py

```diff
@@ -512,80 +544,122 @@ def add_files(self, filenames, target=None):
             log.debug("Adding files %r", new_files)
             new_files.sort(key=lambda x: x.filename)
             if target is None or target is self.unclustered_files:
-                self.unclustered_files.add_files(new_files)
+                #self.unclustered_files.add_files(new_files)
                 target = None
+
             for file in new_files:
-                file.load(partial(self._file_loaded, target=target))
+                self._pending_files[file.filename] = (file, target)
+                self._pending_files_process_queue.put(file.filename)
+
+    def _add_files_thread(self):
+        import time
+        while True:
+            filename, metadata = self._pending_files_results_queue.get()
+
+            if self.stopping:
+                break
+
+            if filename in self._pending_files:
+                file, target = self._pending_files.pop(filename)
+                if target is None or target is self.unclustered_files:
+                    thread.to_main(self.unclustered_files.add_files, [file])
+                    time.sleep(0.005)
```

Comment:
Why is this timer needed?

Reply:
To yield the thread from the time slice. I would use std::this_thread::yield if this was C++.